File: src/main/java/com/heybys/oddments/fooddelivery/persistence/shop/ShopRepositoryImpl.java
package com.heybys.oddments.fooddelivery.persistence.shop;

import org.springframework.stereotype.Repository;

import com.heybys.oddments.base.jpa.BaseRepository;
import com.heybys.oddments.fooddelivery.domain.shop.Shop;
import com.heybys.oddments.fooddelivery.domain.shop.ShopId;
import com.heybys.oddments.fooddelivery.domain.shop.ShopRepository;

@Repository
class ShopRepositoryImpl extends BaseRepository<Shop, ShopId, ShopJpaRepository> implements ShopRepository {

    public ShopRepositoryImpl(ShopJpaRepository repository) {
        super(repository);
    }
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/persistence/shop/ShopRepositoryImpl.kt
package com.heybys.oddments.fooddelivery.persistence.shop

import com.heybys.oddments.base.jpa.BaseRepository
import com.heybys.oddments.fooddelivery.domain.shop.Shop
import com.heybys.oddments.fooddelivery.domain.shop.ShopId
import com.heybys.oddments.fooddelivery.domain.shop.ShopRepository
import org.springframework.stereotype.Repository

@Repository
internal class ShopRepositoryImpl(repository: ShopJpaRepository) :
    BaseRepository<Shop, ShopId, ShopJpaRepository>(repository), ShopRepository {

    override fun getShops(): List<Shop> {
        return repository.findAll()
    }
}
